WcaOnRails/lib/auxiliary_data_computation.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+module AuxiliaryDataComputation
+  def self.compute_everything
+    self.compute_best_of_3_in_333bf
+    self.compute_concise_results
+    self.compute_rank_tables
+
+    self.delete_php_cache # Note: this should go away together with the PHP code.
+  end
+
+  ## Compute mean for 'best of 3' results in 333bf.
+  def self.compute_best_of_3_in_333bf
+    # Set new DNF average where any of three solves is not completed.
+    Result.where(eventId: "333bf", formatId: "3", average: 0)
+          .where("LEAST(value1, value2, value3) < 0")
+          .where.not(value1: 0, value2: 0, value3: 0)
+          .update_all(average: -1)
+    # Set new averages (round times > 10:00).
+    Result.where(eventId: "333bf", formatId: "3", average: 0)
+          .where("LEAST(value1, value2, value3) > 0")
+          .update_all <<-SQL
+            average = IF(
+              (value1 + value2 + value3)/3.0 > 60000,
+              (value1 + value2 + value3)/3.0 - MOD((value1 + value2 + value3)/3.0, 100),
+              (value1 + value2 + value3)/3.0
+            )
+          SQL
+  end
+
+  ## Build 'concise results' tables.
+  def self.compute_concise_results
+    [
+      %w(best ConciseSingleResults),
+      %w(average ConciseAverageResults),
+    ].each do |field, table_name|
+      ActiveRecord::Base.connection.execute "DROP TABLE IF EXISTS #{table_name}"
+      ActiveRecord::Base.connection.execute <<-SQL
+        CREATE TABLE #{table_name}
+        SELECT
+          result.id,
+          #{field},
+          valueAndId,
+          personId,
+          eventId,
+          country.id countryId,
+          continentId,
+          year, month, day
+        FROM (
+            SELECT MIN(#{field} * 1000000000 + result.id) valueAndId
+            FROM Results result
+            JOIN Competitions competition ON competition.id = competitionId
+            JOIN Events event ON event.id = eventId AND event.rank < 990
+            WHERE #{field} > 0
+            GROUP BY personId, result.countryId, eventId, year
+          ) MinValuesWithId
+          JOIN Results result ON result.id = valueAndId % 1000000000
+          JOIN Competitions competition ON competition.id = competitionId
+          JOIN Countries country ON country.id = result.countryId
+      SQL
+    end
+  end
+
+  ## Build rank tables.
+  def self.compute_rank_tables
+    [
+      %w(best RanksSingle ConciseSingleResults),
+      %w(average RanksAverage ConciseAverageResults),
+    ].each do |field, table_name, concise_table_name|
+      ActiveRecord::Base.connection.execute "TRUNCATE TABLE #{table_name}"
+      current_country = Person.current.pluck(:wca_id, :countryId).to_h
```

Comment:
Yeah, I thought hard about a name that doesn't look redundant in use...

But it's different for functions and hashes. A function is declared with a parameter name that tells the reader what's supposed to be passed in, so describing that in the name too is redundant. A caching hash like this _works_ pretty much as a function, but is declared differently, so you have different considerations in naming.